### ISSUE TITLE
`NaN` on `encode_image`

### ISSUE THREAD

**Issue by sarahESL** [openai/CLIP#144]
I fine-tuned CLIP on my own dataset and plan to use it as a pretrained network in order to encode images. After loading the model, I get `NaN` values when using `encode_image` on my images. More interestingly, this happens only since the second batch, so the first batch is alright. I checked the loaded images from the second batch, and they are fine. Any ideas what might be the issue?

**Comment by jongwook** [comment]
Not sure. Are you using the encoder in `.eval()` mode, and did you use `jit=False` (now the default) when loading the model? On the ResNet-based models the batchnorm layers might be outputting unintended results in the train mode.

Casting the entire model to fp32 by `model.float()` or checking if any of the weights have a `NaN` element before and after the `encode_image` call might be helpful.

You can also try checking the output of each layer to see where the `NaN` started to happen during encoding.

**Comment by sarahESL** [comment]
Thanks. Casting the model to fp32 seems to have solved the problem for now.

**Comment by NotNANtoN** [comment]
@sarahESL  Hey, I'm curious about fine-tuning CLIP.

Could you share some details on how many image-text pairs you used and what batch size? Did it work to fine-tune it? I'd appreciate any input! :)

**Comment by sarahESL** [comment]
@NotNANtoN Hey, sure. I used more than 60k image-text pair with batchsizes of 16, 32, 64. Yes, the finetuning worked. I was getting `NaN`s on encoding images in test phase, but then casting the model to fp32 by `model.float()` worked, as suggested by @jongwook.

**Comment by sarahESL** [comment]
Just wanted to document that this issue happens on `encode_text` as well and casting to fp32 is still the fix.
